src/index.d.ts

```diff
@@ -9,54 +9,56 @@ export interface RoutineTypes {
     FULFILL: string;
 }
 
-export type RoutineAction<T> = Action<T>;
-
-export type RoutineActionCreator<T> = (payload?: T) => RoutineAction<T>;
-
-export interface RoutineActionCreators {
-    trigger: RoutineActionCreator<any>;
-    request: RoutineActionCreator<any>;
-    success: RoutineActionCreator<any>;
-    failure: RoutineActionCreator<any>;
-    fulfill: RoutineActionCreator<any>;
+export interface RoutineActionCreators<TActionCreator> {
+    trigger: TActionCreator;
+    request: ActionFunctionAny<Action<any>>;
+    success: ActionFunctionAny<Action<any>>;
+    failure: ActionFunctionAny<Action<any>>;
+    fulfill: ActionFunctionAny<Action<any>>;
 }
 
-export type Routine<T> = RoutineTypes & RoutineActionCreators & T;
+export type Routine<TActionCreator> = TActionCreator & RoutineTypes & RoutineActionCreators<TActionCreator>;
 
 export function routinePromiseWatcherSaga(): Generator;
 
 export function createRoutine(
     typePrefix: string
-): Routine<ActionFunction0<RoutineAction<void>>>;
+): Routine<ActionFunctionAny<Action<any>>>;
```

Comment:
Since `createRoutine` calls `createAction`, a `createAction` _implementation_ (not definition until https://github.com/DefinitelyTyped/DefinitelyTyped/pull/25286) supports `mySimpleAction()`, `mySimpleAction(payload)`, and `mySimpleAction(payload, nothing, interesting, unless, payloadIsError)` for `const mySimpleAction = createAction('SIMPLE_ACTION')`, thus `createRoutine()` returning a `trigger` should behave the same